src/pages/Settings/Network.tsx

```diff
@@ -75,6 +76,9 @@ const Network = (): JSX.Element => {
       // Refresh providers
       await refreshProviders(providers, network, dispatch);
       dispatch(clearAddresses());
+      dispatch(resetUtxos());
```

Comment:
We should prevent to dispatch those, if one is already been dispatched, to save honerous requests.

In Marina we prevented switching networks if an update is not complete.

Reply:
Altough we can do things in background in parallel

Reply:
Checking isFetchingUtxos